doc/config-replace/SONiC_Configuration_Replace_Design_Doc.md

```diff
@@ -0,0 +1,537 @@
+| config replace *filename* best-effort debug        | Move SONiC switch configuration state to user<br/>  specified configuration file and ignore any <br/> errors encountered while performing this operation. <br>Provide debug level details about each step executed as part of the operation. |
+| config replace *filename* best-effort wait-time 20 | Move SONiC switch configuration state to user<br/>  specified configuration file and ignore any <br/> errors encountered while performing this operation. <br/> The configuration patch apply step allows the user specified <br>time interval of 20 seconds  to finish processing of a <br>ConfigDB table update operation. |
+
+#### 3.6.2.2 Show Commands
+
+**Command Format**
+
+show config-replace log [exec | verify | status]
+
+| Command                        | Purpose                                                      |
+| ------------------------------ | ------------------------------------------------------------ |
+| show config-replace log exec   | Displays a log of all the ConfigDB operations executed including<br/>those that failed. In case of a failed operation, it displays an<br/>error message against the failed operation. |
+| show config-replace log verify | Displays a log all the ConfigDB operations that failed,  <br>along with an error message. It does not display the <br/>operations that were successful. |
+| show config-replace log status | Displays the status of last successful configuration replace<br/>operation since switch reboot, including in-progress, successful, <br>and failure. |
+
+
+#### 3.6.2.3 Debug Commands
+
+Use the *debug* command option to view additional details while executing the config replace command.
+
+
+
+# 4 Flow Diagrams
+
+**Configuration Replace Operation**
+![Configuration Replace Flow Diagram](images/config_replace.png)
+
+**ConfigDB TABLE1 SYNC** 
+![ConfigDB Table sync](images/config_db_table_sync.png)
+
+# 5 Error Handling
+
+If an error is encountered during config replace operation, the *config reload* command is executed to  recover from the error state.
+
+# 6 Serviceability and Debug
+The config replace operation logs are stored in systemd-journal and syslog.
+
+# 7 Warm Boot Support
+N/A
+
+# 8 Scalability
+N/A
+
+# 9 Unit Tests
+
+
+
+| Test Case | Description |
+| --------- | ----------- |
+|1|Add a new table.|
+|2|Remove an existing table.|
+|3|Modify values of an existing table entry.|
+|4|Modify value of an existing item an array value.|
+|5|Add a new item to an array  value.|
+|6|Remove an item form an array value.|
+|7|Add a new key to an existing table.|
+|8|Remove a key from an existing  table.|
+|9|Replace the switch configuration from a simple configuration to a configuration file with huge configuration difference. Use maximum number of VLANs and ports to create a huge configuration.|
```

Comment:
Is there any performance impact /downtime while using config replace on day to day operations under traffic? List out if you have any suggestions to use it right?